packages/core/src/utils/prepareEvent.ts

```diff
@@ -0,0 +1,192 @@
+import { ClientOptions, Event, EventHint } from '@sentry/types';
+import { dateTimestampInSeconds, normalize, resolvedSyncPromise, truncate, uuid4 } from '@sentry/utils';
+
+import { Scope } from '../scope';
+
+/**
+ * Adds common information to events.
+ *
+ * The information includes release and environment from `options`,
+ * breadcrumbs and context (extra, tags and user) from the scope.
+ *
+ * Information that is already present in the event is never overwritten. For
+ * nested objects, such as the context, keys are merged.
+ *
+ * Note: This also triggers callbacks for `addGlobalEventProcessor`, but not `beforeSend`.
+ *
+ * @param event The original event.
+ * @param hint May contain additional information about the original exception.
+ * @param scope A scope containing event metadata.
+ * @returns A new event with more information.
+ * @hidden
+ */
+export function prepareEvent(
+  options: ClientOptions,
+  event: Event,
+  hint: EventHint,
+  scope?: Scope,
+): PromiseLike<Event | null> {
+  const { normalizeDepth = 3, normalizeMaxBreadth = 1_000 } = options;
+  const prepared: Event = {
+    ...event,
+    event_id: event.event_id || hint.event_id || uuid4(),
+    timestamp: event.timestamp || dateTimestampInSeconds(),
+  };
+
+  applyClientOptions(prepared, options);
+  applyIntegrationsMetadata(
+    prepared,
+    options.integrations.map(i => i.name),
+  );
+
+  // If we have scope given to us, use it as the base for further modifications.
+  // This allows us to prevent unnecessary copying of data if `captureContext` is not provided.
+  let finalScope = scope;
+  if (hint.captureContext) {
+    finalScope = Scope.clone(finalScope).update(hint.captureContext);
+  }
+
+  // We prepare the result here with a resolved Event.
+  let result = resolvedSyncPromise<Event | null>(prepared);
+
+  // This should be the last thing called, since we want that
+  // {@link Hub.addEventProcessor} gets the finished prepared event.
+  //
+  // We need to check for the existence of `finalScope.getAttachments`
+  // because `getAttachments` can be undefined if users are using an older version
+  // of `@sentry/core` that does not have the `getAttachments` method.
+  // See: https://github.com/getsentry/sentry-javascript/issues/5229
+  if (finalScope) {
```

Comment:
Here a small change: Previously, this was:

```js
if( finalScope && finalScope.getAttachments) {
  // ...
}
```

However, that meant that if a scope existed BUT no `getAttachments`, it would never run `applyToEvent`, which I guess is not desired. cc @AbhiPrasad , or was that on purpose this way?

Reply:
This was to guard against cases where user's had different package versions and `getAttachments` was not defined on the scope object. This could happen if someone was using `captureContext` on a `captureException` call that passed in a old scope. This is on purpose, since then attachments didn't exist as a concept on the scope.

Reply:
Yeah, I get that, what I meant is that we actually guard for _all_ the `finalScope` stuff this way, including this:

```js
result = finalScope.applyToEvent(prepared, hint);
```

Is it desired that this will _not_ be called when `finalScope.getAttachments` is not defined?

Reply:
For future reference: We discussed this internally and IMO we can go with the change and revert this particular part if we get mismatch errors again.

Reply:
Note: We _should_ still be handling mismatches, unless something inside of `scope.applyToEvent` relies on attachments to exists, which I think it shouldn't, as far as I can tell.